pkg/ottl/ottlfuncs/README.md

```diff
@@ -58,6 +59,21 @@ Available Editors:
 - [set](#set)
 - [truncate_all](#truncate_all)
 
+### append
+
+`append(target, Optional[value], Optional[values])`
+
+The `append` function appends single or multiple string values to `target`. 
+`append` converts scalar values into an array if the field exists but is not an array, and creates an array containing the provided values if the field doesn’t exist.
+
+Resulting field is always of type `pcommon.Slice` and will not convert the types of existing or new items in the slice. This means that it is possible to create a slice whose elements have different types.  Be careful when using `append` to set attribute values, as OTLP does not support slices as attribute values when the elements in the slice are not all the same type.
```

Comment:
```suggestion
Resulting field is always of type `pcommon.Slice` and will not convert the types of existing or new items in the slice. This means that it is possible to create a slice whose elements have different types.  Be careful when using `append` to set attribute values, as this will produce values that are not possible to create through OpenTelemetry APIs [according to](https://opentelemetry.io/docs/specs/otel/common/#attribute) the OpenTelemetry specification.
```

>OTLP does not support slices as attribute values when the elements in the slice are not all the same type.

I don't think this is true since the proto supports this (take [log](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L175) or [resource](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/resource/v1/resource.proto#L32) attributes for example). The only reference I see to this is [in the spec](https://opentelemetry.io/docs/specs/otel/common/#attribute) which we shouldn't limit ourselves to.

I have a suggesting for the wording I think we should take here.